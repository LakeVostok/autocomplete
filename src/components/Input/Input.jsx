import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./Input.scss";

export default class Input extends Component {
    static propTypes = {
        value: PropTypes.string,
        placeholder: PropTypes.string,
        onChange: PropTypes.func,
        width: PropTypes.number
    }

    render() {
        let style = {
            width: this.props.width + "px"
        };

        return (
            <input
                value={this.props.value}
                placeholder={this.props.placeholder}
                onChange={this.props.onChange}
                className={styles.input}
                style={style}
            />
        );
    }
}
