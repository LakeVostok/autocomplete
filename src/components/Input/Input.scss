:local {
    .input {
        box-sizing: border-box;
        width: 100%;
        padding: 6px 8px;
        border: 1px solid #d9d9d9;
        border-top-color: #b2b2b2;

        &.error {
            border-color: #d92b2b;
            box-shadow: 0 0 0 1px #d92b2b;
        }

        &:focus {
            outline: 0;
            border-color: #5199db;
            box-shadow: 0 0 0 1px #5199db;

            &::placeholder {
                color: #ccc;
            }
        }

        &::ms-clear {
            display: none;
        }
    }
}