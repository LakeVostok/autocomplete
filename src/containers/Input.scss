.autocomplete__input {
    width: 100%;
    padding: 6px 8px;
    box-sizing: border-box;
    border: 1px solid #d9d9d9;
    border-top-color: #b2b2b2;
}

.autocomplete__input--focused {
    border-color: #5199db;
    box-shadow: 0 0 0 1px #5199db;
}

.autocomplete__input-label {
    position: relative;
}