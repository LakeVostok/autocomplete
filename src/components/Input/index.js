import Input from './Input.jsx';
module.exports = Input;