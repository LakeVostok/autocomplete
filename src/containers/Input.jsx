import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as inputActions from '../actions/inputActions';

import './Input.scss';

class Input extends Component {
    constructor(props) {
        super(props);
    }

    render() {
      let { inputValue, onFocus, onBlur } =  this.props.inputActions;
      let { value, focused } = this.props.input;
      let { id, placeholder } = this.props;
      
      let className = `autocomplete__input ${ focused ? 'autocomplete__input--focused' : '' }`;

      return (
        <label>
          <input className = { className }
            value =          { value.id }
            onChange =       { e => inputValue(id, e) } 
            onFocus =        { e => onFocus(id, e) } 
            onBlur =         { e => onBlur(id) }
            placeholder =    { placeholder }/>
        </label>
      );
    }

}

function mapStateToProps(state) {
  return {
    input: state.input
  };
}

function mapDispatchToProps(dispatch) {
  return {
    inputActions: bindActionCreators(inputActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Input);