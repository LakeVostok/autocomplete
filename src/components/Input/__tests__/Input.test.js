import React from "react";
import renderer from "react-test-renderer";
import Input from "../Input";

describe("Input", () => {
    test("empty", () => {
        const tree = renderer
            .create(<Input />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("with 'error' class", () => {
        const tree = renderer
            .create(<Input error />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("with with the specified width", () => {
        const tree = renderer
            .create(<Input width={100} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
