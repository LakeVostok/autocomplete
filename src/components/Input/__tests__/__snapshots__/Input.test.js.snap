// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input empty 1`] = `
<input
  className="input"
  onChange={undefined}
  placeholder={undefined}
  style={
    Object {
      "width": undefined,
    }
  }
  value={undefined}
/>
`;

exports[`Input with 'error' class 1`] = `
<input
  className="input error"
  onChange={undefined}
  placeholder={undefined}
  style={
    Object {
      "width": undefined,
    }
  }
  value={undefined}
/>
`;

exports[`Input with with the specified width 1`] = `
<input
  className="input"
  onChange={undefined}
  placeholder={undefined}
  style={
    Object {
      "width": 100,
    }
  }
  value={undefined}
/>
`;
